aten/src/ATen/ExpandUtils.cpp

```diff
@@ -149,9 +150,9 @@ std::vector<int64_t> infer_dense_strides(IntArrayRef tensor_sizes, IntArrayRef t
   // all dimensions with 0 stride won't move. This is the same behavior as TensorIterator.
   // eg. Given tensor with size/stride (6, 5, 4, 3, 2)/(6, 0, 120, 0, 1), the initial `perm`
   //     is (4, 3, 2, 1, 0) and the sorted `perm` will be (4, 3, 0, 1, 2)
-  for (int i = 1; i < ndim; ++i) {
-    int dim1 = i;
-    for (int dim0 = i - 1; dim0 >= 0; --dim0) {
+  for (const auto i : c10::irange(1, ndim)) {
+    auto dim1 = i;
+    for (auto dim0 = i - 1; dim0 >= 0; --dim0) {
```

Comment:
If `dim0` is an unsigned type, is this equivalent to endless loop, i.e.:
```
for (auto dim0 = i - 1; ; --dim0) {
```
I assume this change causes numerous segfaults

Reply:
Good catch, thanks. I will convert this one to forward loop.

Reply:
I'd also recommend limiting the change to just `c10::irange` conversions. That way if the logic goes wrong, a rollback will leave the good stuff intact.

Reply:
If `dim1` is modified in the loop, I think that would have changed the number of iterations in the original. `c10::irange` just produces numbers and doesn't respect changes to the iteration variable, so it's possible this loop isn't a good candidate for it.

Reply:
> If dim1 is modified in the loop, I think that would have changed the number of iterations in the original.
No, modify the value of dim1 won't modify the value of i so the iteration number of original loop (outer loop) will not be changed, that is the purpose of have dim1 in the first place I assume.

Reply:
Okay, thanks.